operator/pkg/object/objects_test.go

```diff
@@ -587,3 +587,126 @@ func TestK8sObject_Equal(t *testing.T) {
 		})
 	}
 }
+
+func TestK8sObject_ResolveConflict(t *testing.T) {
+	getK8sObject := func(ystr string) *K8sObject {
+		o, err := ParseYAMLToK8sObject([]byte(ystr))
+		if err != nil {
+			return nil
+		}
+		// Ensure that json data is in sync.
+		o.json, _ = o.JSON()
+		return o
+	}
+
+	cases := []struct {
+		desc string
+		o1   *K8sObject
+		o2   *K8sObject
+	}{
+		{
+			desc: "not applicable kind",
+			o1: getK8sObject(`
```

Comment:
optional: this is fine but in general it's more common in the istio tests to have input strings and transform to object during the test. that way you can also handle errors correctly (no need to panic above).